std/async/pool.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+// From https://github.com/rxaviers/async-pool/blob/master/lib/es7.js.
+// Copyright (c) 2017 Rafael Xavier de Souza http://rafael.xavier.blog.br
+// Licensed under MIT
+
+/**
+ * asyncPool is like Promise.all(array.map(async () => {...})), except that you
+ * can specify the maximum amount of items being processed concurrently.
+ * 
+ * @param poolLimit The maximum count of items being processed concurrently. 
+ * @param array The input array for mapping.
+ * @param iteratorFn The function to call for every item of the array.
+ */
+export async function asyncPool<T, R>(
```

Comment:
Do we need to prefix this with async?

```suggestion
export async function pool<T, R>(
```